infra/terraform/vpc_hub/README.md

```diff
@@ -0,0 +1,73 @@
+# VPC Hub
+
+This Terraform Module creates a VPC Hub for VPC peering connection to VPC Spokes. The goal of this VPC is to centralize VPC PrivateLink connections to AWS services so we don't have to create VPC PrivateLink connections in each VPC Spoke for all services we use. This will allow us to only pay for the VPC Interface Endpoints in the VPC Hub once.
+
+A VPC Spoke is simply just another VPC which wants to access AWS services without going through the public internet.
+
+```mermaid
+graph LR
+    S[AWS Service]
+    H[VPC Hub]
+    V1[VPC Spoke X]
+    V2[VPC Spoke Y]
+
+    H -- PrivateLink <--> S
+    V1 -- VPC Peering Connection <--> H
+    V2 <--> H
+
+```
+
+## Considerations for using the VPC Hub
+
+When a VPC Spoke is to be connected to the Hub, it's essential that theres a non-ambiguous network route to the Spoke. Essentially this means our infrastructure should have unique CIDR blocks for each environment in every project. If this is not the case, the VPC Hub will not know which Spoke to route traffic back to. I suggest reading this [AWS documentation site](https://docs.aws.amazon.com/vpc/latest/peering/peering-configurations-partial-access.html#peering-incorrect-response-routing) to understand why this is the case.
```

Comment:
What do we mean by 'environment' here? I understand that 'project' corresponds to a VPC (I think)

Reply:
An environment would be `dev`, `prod`, `test`, etc. We currently only use prod/dev, but the given delegation scheme allows up to 4 environments per project. For example, `HASH` would be the project, and `prod` would be an environment we host in a VPC